src/core/continuous.jl

```diff
@@ -284,7 +284,10 @@ function SciMLBase.reinit!(integ::AbstractODEIntegrator, u0::AbstractVector,
     reinit!(integ, integ.u; kwargs...)
 end
 
-successful_step(integ) = true
+function successful_step(integ)
+	all(isfinite,integ.u) ? true : false
```

Comment:
We need to use `any` here. If any entry of the vector is infinite we already have a problem. `all` doesn't cover this.

Also, please use proper API, i.e., `get_state(integ)` instead of `integ(u)`.

Reply:
Nono, `all(isfinite,integ.u)` checks if **all** of them are **finite** , so we are on the same page. `any(isinfinite,integ.u)` would be the same. I'll correct the state part though.

Reply:
Aaaaah sorry, my apologies, I missread your `isfinite` for `isinfinite`. Great :)